Sources/StreamChat/ChatClient.swift

```diff
@@ -483,15 +483,33 @@ public class ChatClient {
         }
     }
     
-    /// Disconnects the chat client form the chat servers and removes all the local data related.
+    /// Disconnects the chat client from the chat servers and removes all the local data related.
     @available(*, deprecated, message: "Use the asynchronous version of `logout` for increased safety")
     public func logout() {
         logout {}
     }
 
     /// Disconnects the chat client from the chat servers and removes all the local data related.
-    public func logout(completion: @escaping () -> Void) {
-        authenticationRepository.logOutUser()
+    /// - Parameters:
+    ///  - removeDevice: If `true`, it removes the current device from the user's registered devices automatically.
+    ///  By default it is enabled.
+    public func logout(
+        removeDevice: Bool = true,
```

Comment:
Should we go with `false` initially? Since it's a breaking change, and many customers would receive "device not exists errors" when they do their request.

Reply:
The removeDevice is called only if it was not done before. I've tested this, and with the old demo app code, this additional call is not performed because currentDevice has been deleted already. It is stated in the PR description:

> However, if a customer has already deleted the user device before calling logout, it won't call logout again.

Reply:
ah cool, missed that

Reply:
But if we want to minimize the risk as much as possible we could set this flag to `false`. But the problem is that this will keep increasing the support requests, as well as require updating the documentation that this flag needs to be set in order for the remove device to be executed. On Android, this is also done by default, so it would be weird if had this different from them. That is why I tried as much as possible to make this enabled by default, without breaking existing integrations